commit 63d37be6b4a854b9cbbd47306c5c383a2ad356c2
Author: Panu Matilainen <pmatilai@redhat.com>
Date:   Tue Sep 15 12:57:51 2009 +0300

    Initialize rpm earlier in rpm2cpio (RhBug:523260)
    - url retrieval requires macros to be loaded

diff --git a/rpm2cpio.c b/rpm2cpio.c
index 6c29cab..bf7f17c 100644
--- a/rpm2cpio.c
+++ b/rpm2cpio.c
@@ -21,6 +21,7 @@ int main(int argc, char *argv[])
     FD_t gzdi;
     
     setprogname(argv[0]);	/* Retrofit glibc __progname */
+    rpmReadConfigFiles(NULL, NULL);
     if (argc == 1)
 	fdi = fdDup(STDIN_FILENO);
     else {
@@ -37,7 +38,6 @@ int main(int argc, char *argv[])
 	exit(EXIT_FAILURE);
     }
     fdo = fdDup(STDOUT_FILENO);
-    rpmReadConfigFiles(NULL, NULL);
 
     {	rpmts ts = rpmtsCreate();
 	rpmVSFlags vsflags = 0;
